starboard/nplb/player_test_fixture.cc

```diff
@@ -178,6 +183,66 @@ void SbPlayerTestFixture::Write(const AudioSamples& audio_samples,
   }
 }
 
+void SbPlayerTestFixture::Write(SbTime start, SbTime end) {
+  SB_DCHECK(thread_checker_.CalledOnValidThread());
+  SB_DCHECK(SbPlayerIsValid(player_));
+  SB_DCHECK(!audio_end_of_stream_written_);
+  SB_DCHECK(!video_end_of_stream_written_);
+
+  ASSERT_FALSE(error_occurred_);
+  ASSERT_TRUE(dmp_demuxer_->SetReadRange(start, end));
+
+  bool has_more_audio = dmp_demuxer_->HasMediaTrack(kSbMediaTypeAudio);
+  bool has_more_video = dmp_demuxer_->HasMediaTrack(kSbMediaTypeVideo);
+
+  int max_audio_samples_per_write =
+      SbPlayerGetMaximumNumberOfSamplesPerWrite(player_, kSbMediaTypeAudio);
+  int max_video_samples_per_write =
+      SbPlayerGetMaximumNumberOfSamplesPerWrite(player_, kSbMediaTypeVideo);
+
+  while (has_more_audio || has_more_video) {
+    ASSERT_NO_FATAL_FAILURE(WaitForDecoderStateNeedsData());
+    if (can_accept_more_audio_data_) {
+      int start_index = dmp_demuxer_->GetCurrentIndex(kSbMediaTypeAudio);
+      int audio_samples_to_write = 0;
+      SbPlayerSampleInfo sample_info;
+      while (dmp_demuxer_->ReadSample(kSbMediaTypeAudio, sample_info)) {
+        ++audio_samples_to_write;
+        if (audio_samples_to_write == max_audio_samples_per_write) {
+          break;
+        }
+      }
+      if (audio_samples_to_write > 0) {
+        ASSERT_NO_FATAL_FAILURE(WriteSamples(kSbMediaTypeAudio, start_index,
```

Comment:
We iterate through the DmpDemuxer above, then we write the samples using DmpReaders.  We need to clarify the responsibility of DmpDemuxer?  Does it only provide indices?